src/lib/gadgets/lookup.unit-test.ts

```diff
@@ -0,0 +1,57 @@
+import { mod } from '../../bindings/crypto/finite_field.js';
+import { Field } from '../../lib/core.js';
+import { ZkProgram } from '../proof_system.js';
+import {
+  Spec,
+  boolean,
+  equivalentAsync,
+  fieldWithRng,
+} from '../testing/equivalent.js';
+import { Random } from '../testing/property.js';
+import { assert } from './common.js';
+import { Gadgets } from './gadgets.js';
+import { constraintSystem, contains } from '../testing/constraint-system.js';
+
+let maybeUint = (n: number | bigint): Spec<bigint, Field> => {
+  let uint = Random.bignat((1n << BigInt(n)) - 1n);
+  return fieldWithRng(
+    Random.map(Random.oneOf(uint, uint.invalid), (x) => mod(x, Field.ORDER))
+  );
+};
```

Comment:
I originally misunderstood this. It just returns a value that's sometimes less than `n` bits and sometimes more, correct? Its not doing anything else?

Reply:
Yes